piksi_tools/console/windows_scaling.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2019 Swift Navigation Inc.
+# Contact: engineering@swiftnav.com
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+import sys
+import subprocess
+import pyface.qt
+
+# A hack to use Windows legacy scaling mode when started on a high dpi screen
+# using Qt5. Otherwise the layout would change somewhat, and plot legend fonts
+# would become too small. The downside can be some blurriness due to scaling.
+# See https://doc.qt.io/qt-5/highdpi.html
+#
+# Ideally would become unnecessary one day when all components, plots included,
+# would properly work in a high dpi environment.
+if sys.platform == "win32" and pyface.qt.qt_api.lower() in ['pyqt5', 'pyside2']:
+    prog_str = b"""\
+from PyQt5 import QtWidgets
+
+app = QtWidgets.QApplication([])
+dpi_list = [screen.logicalDotsPerInch() for screen in app.screens()]
+max_dpi = max(dpi_list)
+
+print(max_dpi, end="")
+"""
+    # initializing the application to use this Windows scaling mechanism is
+    # only possible before initializing the QApplication for the first time,
+    # so the dpi query, which also uses Qt, is done in a separate process
+    helper_proc = subprocess.Popen([sys.executable, "-u", "-"],
```

Comment:
I think once piksi_tools is reduced to a single executable by PyInstaller `sys.executable` points to `console.exe`-- I'm currently seeing that this logic "fork bombs" the computer by repeatedly launching `console.exe`.  Which results in error dialogs like this:

![low-dpi-scaling-fix-error](https://user-images.githubusercontent.com/183436/56329457-70cc2280-6138-11e9-9181-5c5bcbb49265.PNG)

![low-dpi-scaling-fix-error2](https://user-images.githubusercontent.com/183436/56329461-74f84000-6138-11e9-930a-864d9c7d4449.PNG)